### PR TITLE
Migrate recipe and attributes from jenkins role to recipe for testing

### PR DESCRIPTION
**Pull request by armiller**
This also updates java to openjdk 8. I have verify it worked within the test kitchen vm.

### CODE REVIEW

File: recipes/master.rb [hunk 1]
```diff
@@ -25,3 +25,23 @@
...
+} ]
+
+include_recipe 'java'
+include_recipe 'jenkins::master'
+include_recipe 'certificate::manage_by_attributes'
+include_recipe 'osl-jenkins::haproxy'
+include_recipe 'osl-haproxy::default'
+include_recipe 'git'
+include_recipe 'osl-jenkins::chef_ci_cookbook_template'
+include_recipe 'python'
```
Comment (ramereth) [inline recipes/master.rb]: Shouldn't this be ``poise-python`` now?

File: .kitchen.yml [hunk 2]
```diff
@@ -11,8 +11,15 @@ suites:
...
-      - recipe[jenkins::master]
+    attributes:
+      certificate:
+        - snakeoil:
+            cert_path: "/etc/pki/tls"
+            cert_file: wildcard.pem
+            key_file: wildcard.key
+            chain_file: wildcard-bundle.crt
+            combined_file: true
+
```
Comment (ramereth) [inline .kitchen.yml]: You should remove this since you reference it in ``master.rb``.

Comment (armiller) [inline .kitchen.yml]: this is needed for testing.

**Comment by ramereth** [comment]
:

**Pull request merged**